pallets/messages/src/benchmarking.rs

```diff
@@ -77,6 +80,45 @@ benchmarks! {
 		}
 	}: _ (RawOrigin::Signed(caller), Some(message_source_id.into()), schema_id, payload)
 
+	add_onchain_message_too_large_payload {
+		let n = T::MaxMessagePayloadSizeBytes::get() + 1;
+		let message_source_id = DelegatorId(2);
+		let caller: T::AccountId = whitelisted_caller();
+		let schema_id = 1;
+		let payload = vec![1; n as usize];
+
+	}: {
+		assert_err_ignore_postinfo!(MessagesPallet::<T>::add_onchain_message(RawOrigin::Signed(caller).into(), Some(message_source_id.into()), schema_id, payload), Error::<T>::ExceedsMaxMessagePayloadSizeBytes);
+	}
+
+	add_onchain_message_invalid_schema {
+		let n in 0 .. T::MaxMessagePayloadSizeBytes::get() - 1;
+		let m in 1 .. MESSAGES;
+		let message_source_id = DelegatorId(2);
+		let caller: T::AccountId = whitelisted_caller();
+		let schema_id = 65534;
+		let payload = vec![1; n as usize];
+	}: {
+		assert_err_ignore_postinfo!(MessagesPallet::<T>::add_onchain_message(RawOrigin::Signed(caller).into(), Some(message_source_id.into()), schema_id, payload), Error::<T>::InvalidSchemaId);
+	}
+
+	add_onchain_message_invalid_account {
```

Comment:
I like this idea. However, I think it would be ok to add one additional error and not 3.